Decorator/GuzzleClientFactoryDecorator.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Auxmoney\OpentracingGuzzleBundle\Decorator;
+
+
+use Auxmoney\OpentracingGuzzleBundle\Middleware\GuzzleRequestSpanning;
+use Auxmoney\OpentracingGuzzleBundle\Middleware\GuzzleTracingHeaderInjection;
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use Symfony\Component\DependencyInjection\Reference;
+
+final class GuzzleClientFactoryDecorator
+{
+
+    private $decorated;
+
+    public function __call($method, $args)
```

Comment:
Does this also intercept other methods like `setDependency()`? Can this be an issue for setter injection?

Can't we get the original information of the factory (in the compiler pass, from the definition) and pass it in here, so that only this method is intercepted by `__call`?

Reply:
Oh, scrolling down revealed, that you already have that information: its `callable $factory` in `addMiddlewareToClientFactory()`.